### PR TITLE
fix: make enum lookup case insensitive

### PR DESCRIPTION
**Pull request by joshgummersall**
I tested the HTTP action in Composer with the .NET runtime, and it appears that enum resolution is case insensitive, so this PR replicates that behavior. I tested this by linking against a Composer bot and manually editing the enum value in the dialog to different casing variations.

Fixes https://github.com/microsoft/BotFramework-Composer/issues/7834
Fixes https://github.com/microsoft/botbuilder-js/issues/3744

### CODE REVIEW

**Comment by joshgummersall** [comment]
@stevengum, as far as I can tell, Dotnet explicitly sets case insensitive here: https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/AdaptiveExpressions/ExpressionProperties/EnumExpression.cs#L104

File: libraries/adaptive-expressions/src/converters/enumExpressionConverter.ts [hunk 1]
```diff
@@ -15,31 +15,45 @@ type Input<T> = T | string | Expression;
...
-                return new EnumExpression<T>(this._enumValue[value as string]);
+
+        if (typeof value === 'string') {
+            let enumValue = this.enumValue[value];
+            if (enumValue === undefined) {
+                enumValue = this.enumValue[this.lowercaseIndex[value]];
+            }
+
+            if (enumValue !== undefined) {
+                return new EnumExpression<T>(enumValue as Input<T>);
```
Comment (stevengum) [inline libraries/adaptive-expressions/src/converters/enumExpressionConverter.ts:51]:
```suggestion
        const enumValue = this.enumValue[value] ?? this.enumValue[this.lowercaseIndex[value]];

        if (enumValue !== undefined) {
            return new EnumExpression<T>(enumValue as Input<T>);
```

**Review by stevengum — CHANGES_REQUESTED** [review]
One small ask and a suggestion for cleanup. Otherwise LGTM

File: libraries/adaptive-expressions/src/converters/enumExpressionConverter.ts (hunk 1 above)
Comment (joshgummersall) [reply libraries/adaptive-expressions/src/converters/enumExpressionConverter.ts:51]: Sadly adaptive expressions is pinned on and older Typescript version. That was my first try!

Comment (stevengum) [reply libraries/adaptive-expressions/src/converters/enumExpressionConverter.ts:51]: That’s a bummer… A dependent’s typescript dependency, right?

**Review by stevengum — APPROVED** [review]

File: libraries/adaptive-expressions/src/converters/enumExpressionConverter.ts (hunk 1 above)
Comment (joshgummersall) [reply libraries/adaptive-expressions/src/converters/enumExpressionConverter.ts:51]: Yep, I don't think our version of antlr works with newer Typescript versions.

**Pull request merged**